5.10.0 Release Notes (#2971)
emqtt/emq-docs-cn: en_US/changes/changes-ee-v5.md

  Previous event topics are kept for backwards compatibility.

- [#15175](https://github.com/emqx/emqx/pull/15175) Added support for matching event topics in Rule Engine using wildcards.  Now, it's possible to use `$events/#`, `$events/sys/+` and similar for matching multiple events at once.

#### Smart Data Hub

- [#15174](https://github.com/emqx/emqx/pull/15174) Added support to upload Protobuf source file bundles for Schema Registry.

  For example, assuming that the Protobuf source file bundle is at `/tmp/bundle.tar.gz` and has the following file structure, with `a.proto` being the root Protobuf schema file:

  ```
  .
  ├── a.proto
  ├── c.proto
  └── nested
      └── b.proto
  ```

  Then, to create a new schema using that bundle via the HTTP API:

  ```sh
  curl -v http://127.0.0.1:18083/api/v5/schema_registry_protobuf/bundle \
    -XPOST \
    -H "Authorization: Bearer xxxx" \
    -F bundle=@/tmp/bundle.tar.gz \
    -F name=my_cool_schema \
    -F root_proto_file=a.proto
  ```

#### Data Integration

- [#15248](https://github.com/emqx/emqx/pull/15248) EMQX supports data integration with [Doris](https://doris.apache.org/), supporting data writing using SQL statements.

> - [#15218](https://github.com/emqx/emqx/pull/15218) Added support for MSK IAM authentication for all Kafka Connectors (Kafka Producer, Kafka Consumer, Confluent, Azure Event Hub).

Reviewer: Not sure if confluent hosted Kafka supports MSK IAM auth, but Azure Event Hubs is most probably not possible.
```suggestion
- [#15218](https://github.com/emqx/emqx/pull/15218) Added support for MSK IAM authentication for Kafka Connectors (Kafka Producer, Kafka Consumer).
```

- [#15160](https://github.com/emqx/emqx/pull/15160) Added the `DELETE /mt/bulk_delete_ns` API for multi-tenancy management, which allows deleting namespaces in bulk.

#### CLI

- [#15158](https://github.com/emqx/emqx/pull/15158) Added new `emqx ctl conf remove x.y.z` command, which removes the configuration key path `x.y.z` from the existing configuration.

#### Gateway

- [#15138](https://github.com/emqx/emqx/pull/15138) Introduced NATS Gateway for accepting NATS client connections over TCP/TLS, WS/WSS transport protocols.

  For example, the NATS gateway will transform the following NATS message into an MQTT message with the topic `sub/t` and payload `hello`, while supporting seamless integration with existing EMQX features such as the rule engine, data integration, and more:
  ```
  PUB sub.t 5  
  hello
  ```

#### Durable Storage

- [#15043](https://github.com/emqx/emqx/pull/15043) Instrument the DS Raft backend with basic metrics to provide insights into cluster status, database overview, shard replication, and replica transitions.

### Bug Fixes

#### Access Control

> - [#15184](https://github.com/emqx/emqx/pull/15184) Fixed incorrect error message format when blacklist creation fails.

Reviewer: ```suggestion
- [#15184](https://github.com/emqx/emqx/pull/15184) Fixed incorrect error message format when a new record creation fails for banned list.
```

#### Clustering

- [#15304](https://github.com/emqx/emqx/pull/15304) Fixed the problem related to core node discovery by replicant nodes when using `static` discovery strategy.

  Previously, the replicants could ignore core nodes not explicitly listed in the `static_seeds` list.
  This could lead to an inconsistent cluster view and load imbalance.

- [#15180](https://github.com/emqx/emqx/pull/15180)Fixed an issue in `ekka_locker` where RPC (`badrpc`) errors were not handled correctly, causing false-positive lock successes. This could lead to inconsistent lock states and deadlocks in clustered deployments.

#### Security

> - [#15159](https://github.com/emqx/emqx/pull/15159) Now, after a CRL Distribution Point URL fails to be refreshed after a number of consecutive attempts (60 s by default), it'll be evicted and stop being refreshed, instead of flooding logs.

Reviewer: ```suggestion
- [#15159](https://github.com/emqx/emqx/pull/15159) Now, after a CRL Distribution Point URL fails to be refreshed after a number of consecutive attempts (60 seconds by default), it'll be evicted and stop being refreshed, instead of flooding logs.
```
